rtsp.go

```diff
@@ -114,53 +116,61 @@ func (rc *rtspCamera) clientReconnectBackgroundWorker(codecInfo videoCodec) {
 					errors.Is(err, io.EOF) ||
 					errors.Is(err, syscall.EPIPE) ||
 					errors.Is(err, syscall.ECONNREFUSED)) {
-					rc.logger.Warnw("The rtsp client encountered an error, trying to reconnect", "url", rc.u, "error", err)
+					rc.logger.Warnf("The rtsp client encountered an error, trying to reconnect to %s, err: %s", rc.u, err)
 					badState = true
 				} else if res != nil && res.StatusCode != base.StatusOK {
-					rc.logger.Warnw("The rtsp server responded with non-OK status", "url", rc.u, "status code", res.StatusCode)
+					rc.logger.Warnf("The rtsp server responded with non-OK status url: %s, status_code: %s", rc.u, res.StatusCode)
 					badState = true
 				}
 			}
 
 			if badState {
 				if err := rc.reconnectClient(codecInfo); err != nil {
-					rc.logger.Warnw("cannot reconnect to rtsp server", "error", err)
+					rc.logger.Warnf("cannot reconnect to rtsp server err: %s", err.Error())
 				} else {
-					rc.logger.Infow("reconnected to rtsp server", "url", rc.u)
+					rc.logger.Infof("reconnected to rtsp server url: %s", rc.u)
 				}
 			}
 		}
 	}, rc.activeBackgroundWorkers.Done)
 }
 
-func (rc *rtspCamera) closeConnection() error {
-	var err error
+func (rc *rtspCamera) closeConnection() {
 	if rc.client != nil {
-		err = rc.client.Close()
+		rc.client.Close()
 		rc.client = nil
 	}
 	if rc.rawDecoder != nil {
 		rc.rawDecoder.close()
 		rc.rawDecoder = nil
 	}
-	return err
 }
 
 // reconnectClient reconnects the RTSP client to the streaming server by closing the old one and starting a new one.
-func (rc *rtspCamera) reconnectClient(codecInfo videoCodec) (err error) {
-	rc.logger.Warnf("reconnectClient called")
-
+func (rc *rtspCamera) reconnectClient(codecInfo videoCodec) error {
 	if rc == nil {
 		return errors.New("rtspCamera is nil")
 	}
 
-	err = rc.closeConnection()
-	if err != nil {
-		rc.logger.Debugw("error while closing rtsp client:", "error", err)
-	}
+	rc.logger.Warnf("reconnectClient called with codec: %s", codecInfo)
+
+	rc.closeConnection()
 
 	// replace the client with a new one, but close it if setup is not successful
 	rc.client = &gortsplib.Client{}
+	rc.client.OnPacketLost = func(err error) {
+		rc.logger.Debugf("OnPacketLost: err: %s", err.Error())
+	}
+	rc.client.OnTransportSwitch = func(err error) {
+		rc.logger.Debugf("OnTransportSwitch: err: %s", err.Error())
```

Comment:
Have we been able to test transport switches? Should we fire off a reconnect in this situation?

Reply:
I have not seen this triggered no